src/momento/config/middleware/aio/middleware_metadata.py

```diff
@@ -0,0 +1,14 @@
+from typing import Optional
+
+from grpc.aio import Metadata
+
+
+class MiddlewareMetadata:
+    """Wrapper for gRPC metadata."""
+
+    def __init__(self, metadata: Optional[Metadata]):
+        self.grpc_metadata = metadata
+
+    def get_grpc_metadata(self) -> Optional[Metadata]:
```

Comment:
Accessors like this are uncommon. Can make `grpc_metadata` private by prefixing with an underscore then make a property if you want to hide it. Or if you go with the dataclass route then you just leave it public.